bottlecap/src/main.rs

```diff
@@ -156,23 +165,76 @@ fn main() -> Result<()> {
                 shutdown_reason,
                 deadline_ms,
             }) => {
                 println!("Exiting: {}, deadline: {}", shutdown_reason, deadline_ms);
-                dogstats_client.shutdown();
-                telemetry_listener.shutdown();
-                event_bus.shutdown();
-                return Ok(());
+                shutdown = true;
             }
             Err(err) => {
                 eprintln!("Error: {:?}", err);
                 println!("Exiting");
                 return Err(err);
             }
         }
+        // Block until we get something from the telemetry API
+        // Check if flush logic says we should block and flush or not
+        if flush_control.should_flush() {
+            loop {
+                let received = event_bus.rx.recv();
+                if let Ok(event) = received {
+                    match event {
+                        Event::Metric(event) => {
+                            debug!("Metric event: {:?}", event);
+                        }
+                        Event::Telemetry(event) => {
+                            match event.record {
+                                TelemetryRecord::PlatformInitReport {
+                                    initialization_type,
+                                    phase,
+                                    metrics,
+                                } => {
+                                    debug!("Platform init report for initialization_type: {:?} with phase: {:?} and metrics: {:?}", initialization_type, phase, metrics);
+                                    // write this straight to metrics aggr
+                                    // write this straight to logs aggr
+                                    // write this straight to trace aggr
+                                }
+                                TelemetryRecord::PlatformRuntimeDone {
+                                    request_id, status, ..
+                                } => {
+                                    debug!(
+                                        "Runtime done for request_id: {:?} with status: {:?}",
+                                        request_id, status
+                                    );
+                                    debug!("FLUSHING ALL");
+                                    dogstats_client.flush();
+                                    debug!("CALLING FOR NEXT EVENT");
+                                    break;
+                                }
+                                TelemetryRecord::PlatformReport {
+                                    request_id, status, ..
+                                } => {
+                                    debug!(
+                                        "Platform report for request_id: {:?} with status: {:?}",
+                                        request_id, status
+                                    );
+                                }
+                                _ => {}
```

Comment:
let's add an warn/error log to report any unsupported event?

Reply:
I'll just make this a debug for now because I expect we move them up into the match as we further develop things.